### PR TITLE
Only use default transitions for non-clear opaque backgrounds

### PR DESCRIPTION
**Pull request by daehn**
## What's new in this PR?

* Only use default transitions when the background is neither clear nor translucent.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Components/Controllers/ClearBackgroundNavigationController.swift [hunk 1]
```diff
@@ -86,7 +86,9 @@ class ClearBackgroundNavigationController: UINavigationController {
     }
     
     func navigationController(_ navigationController: UINavigationController, didShow viewController: UIViewController, animated: Bool) {
-        useDefaultPopGesture = fabs(viewController.view.backgroundColor?.alpha ?? 1.0 - 1.0) < CGFloat.ulpOfOne
+        let clearBackround = viewController.view.backgroundColor == .clear
+        let opaqueBackground = fabs(viewController.view.backgroundColor?.alpha ?? 1.0 - 1.0) < CGFloat.ulpOfOne
+        useDefaultPopGesture = !clearBackround && opaqueBackground
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Components/Controllers/ClearBackgroundNavigationController.swift]: I am not fully following, if background is .clear, should it have also alpha == 0, so then color.alpha would be != 1?

Comment (daehn) [reply Wire-iOS/Sources/UserInterface/Components/Controllers/ClearBackgroundNavigationController.swift]: If the alpha is clear, or the background has an alpha value lower than 1 we should use the default gesture. Before it was only checking whether or not the alpha was lower than 1, which might not be the case for clear background colours.

Comment (daehn) [reply Wire-iOS/Sources/UserInterface/Components/Controllers/ClearBackgroundNavigationController.swift]: `opaqueBackground` should be named `translucentBackground` here though

Comment (mikeger) [reply Wire-iOS/Sources/UserInterface/Components/Controllers/ClearBackgroundNavigationController.swift]: So there can be a clear color that would not match "alpha < 1" ?

Comment (daehn) [reply Wire-iOS/Sources/UserInterface/Components/Controllers/ClearBackgroundNavigationController.swift]: You're right, I think I misread this, I thought we were checking the view's alpha, but we are already checking the color's alpha! Good point, will have a look again.

Comment (daehn) [reply Wire-iOS/Sources/UserInterface/Components/Controllers/ClearBackgroundNavigationController.swift]: @mikeger fixed now!

**Pull request merged**